app/models/folder/Folder.scala

```diff
@@ -0,0 +1,307 @@
+
+  def parse(r: FoldersRow): Fox[Folder] =
+    Fox.successful(Folder(ObjectId(r._Id), r.name))
+
+  def parseWithParent(t: (String, String, Option[String])): Fox[FolderWithParent] =
+    Fox.successful(FolderWithParent(ObjectId(t._1), t._2, t._3.map(ObjectId(_))))
+
+  override def readAccessQ(requestingUserId: ObjectId): String = readAccessQWithPrefix(requestingUserId, "")
+
+  def readAccessQWithPrefix(requestingUserId: ObjectId, prefix: String): String =
+    rawAccessQ(write = false, requestingUserId, prefix)
+
+  override def updateAccessQ(requestingUserId: ObjectId): String =
+    rawAccessQ(write = true, requestingUserId, prefix = "")
+
+  private def rawAccessQ(write: Boolean, requestingUserId: ObjectId, prefix: String): String = {
+    val writeAccessPredicate = if (write) "AND tr.isTeamManager" else ""
+    val breadCrumbsAccessForFolder =
+      if (write) ""
+      else
+        s"""
+       -- only for read access: you may read the ancestors of a folder you can access
+       UNION ALL
+       SELECT fp._ancestor
+       FROM webknossos.folder_paths fp
+       WHERE fp._descendant IN (
+         SELECT at._folder
+         FROM webknossos.folder_allowedTeams at
+         JOIN webknossos.user_team_roles tr ON at._team = tr._team
+         WHERE tr._user = '$requestingUserId'
+       )
+       """
+    val breadCrumbsAccessForDataset =
+      if (write) ""
+      else
+        s"""
+       -- only for read access: you may read the ancestors of a dataset you can access
+       UNION ALL
+       SELECT fp._ancestor
+       FROM webknossos.folder_paths fp
+       WHERE fp._descendant IN (
+         SELECT d._folder
+         FROM webknossos.dataSets_ d
+         JOIN webknossos.dataSet_allowedTeams dt ON dt._dataSet = d._id
+         JOIN webknossos.user_team_roles ut ON dt._team = ut._team
+         WHERE ut._user = '$requestingUserId'
+       )
+       """
+    s"""
+        (
+          -- is descendant of user organization folder and user is admin or dataset manager
+          ${prefix}_id IN (
+            SELECT fp._descendant
+            FROM webknossos.folder_paths fp
+            WHERE fp._ancestor IN (
+               SELECT o._rootFolder
+               FROM webknossos.organizations_ o
+               JOIN webknossos.users_ u ON u._organization = o._id
+               WHERE u._id = '$requestingUserId'
+               AND (
+                 u.isAdmin
+                 OR u.isDatasetManager
+               )
+            )
+          )
+        )
+        OR (
+          -- is descendant of a folder with allowed teams the user is in
+          ${prefix}_id IN (
+            SELECT fp._descendant
+            FROM webknossos.folder_paths fp
+            WHERE fp._ancestor IN (
+              SELECT at._folder
+              FROM webknossos.folder_allowedTeams at
+              JOIN webknossos.user_team_roles tr ON at._team = tr._team
+              WHERE tr._user = '$requestingUserId'
+              $writeAccessPredicate
```

Comment:
```suggestion
              AND $writeAccessPredicate
```
(change above to or("true"))